app/views/observations/namings/edit.js.erb

```diff
@@ -0,0 +1,38 @@
+$('body').append('<%= j render(partial: "observations/namings/modal", locals: { action: :update, title: :edit_naming_title.t(id: @observation.id) }) %>');
+
+$('#modal_naming').on('hidden.bs.modal', function () {
+  $(this).remove();
+});
+
+// attach all the bindings on shown
+$('#modal_naming').on('shown.bs.modal', function() {
+  var naming_input = $("#naming_name");
+  var naming_form = $("#naming_form");
+  var namings_table = $("#namings_table");
+  var observation_namings = $("#observation_namings");
+  var this_modal = $(this);
+
+  new MOAutocompleter(jQuery.extend({
+    input_elem: naming_input,
+    ajax_url: "/ajax/auto_complete/name/@",
+    collapse: 1
```

Comment:
So there's a trick you can do with the MO autocompleter that allows you to establish a single name autocompleter that can be used with multiple name fields.  (I developed it for the species list bulk editor which I think is now defunct, but the code should still be sound.)  The concern is that each autocompleter instance loads in *a lot* of names.  Modern browsers use a lot of memory, sure, and can probably handle it... but why not have all the modals share the same autocompleter?

Hmmm.  There it is, not documented, alas.  Check the "reuse" method.  I think I get it.  I think you bind an MOAutocompleter to the first field like usual, then for all subsequent ones you do `$('#first_field').reuse($('#second_field'))` and so on.  This binds a bunch of events on #second_field to switch the input element over to the second one, etc.  It basically rebinds the autocompleter on the fly to whichever field you click into.  It worked well, as I recall.

Reply:
Thanks! I did notice that, and wanted to use it. 

The trick is... the fields have the same ID. The first form is `remove()`d from the DOM, and then the second form is appended, and... there's nothing to reuse.

So I guess what I should do is this:
- Change the `id` attribute of the _old_ field, to `naming_name_old` or something.
- Hide the old form
- reuse the autocompleter `$('#naming_name').reuse($('#naming_name_old'))`
- _Then,_ remove the old form

Do you reckon that would work?

Reply:
Oh. My comment related to `form_reload.js.erb` and you were commenting on `edit.js.erb`.

With `edit.js.erb` and `new.js.erb` there's no name autocompleter to reuse (we're on the show_obs page)... is there?

Reply:
More background... 

The partial that renders the naming form into the modal already has a `turn_into_name_auto_completer`, and I think that should work, but it does not. (Maybe because the modal is still hidden at this point?):
```ruby
        <%= f_n.label(:name, :WHAT.t + ":") %>
        <%= f_n.text_field(:name, value: what, size: 40, class: "form-control",
                           autocomplete: "off", # turn off browser autocomplete
                           data: { autofocus: focus_on_name }) %>
        <% turn_into_name_auto_completer(:naming_name, primer: Name.primer) %>
```
In any case, i found that it didn't work; I had to manually bind the autocompleter with the response template (`new.js.erb` etc) after the modal is shown.

Reply:
I think there is, even on the show obs page.  Each time you propose a name or edit an existing one, it will popup this naming modal, right?  Unless it's the identical modal in each case.  (Sorry, I could check that myself if I were more patient.)  But certainly on the index this will become an issue.

How about this solution?  Create a hidden field, and bind a name autocompleter to it!  Then "reuse" it for all name fields, wherever they appear, whether in the initial page load or in the results of any ajax request.  (Unfortunately the way the autocompleter I adapted worked, it *had to* have a jquery element to work off of.  It's too much work to change that to work off of an "autocompleter" instance instead.)

Reply:
re: why turn_into_autocompleter doesn't work in ajax response -- I believe it's doing an "inject javascript at end" thingy.  That will never happen in an ajax response!

Reply:
Right. Copying from a hidden one already in the page is a nice idea, i like it. 

Will add that field to observations index and show and adjust this. Is the correct syntax:
```js
AUTOCOMPLETERS[jQuery('##{first_id}').data('uuid')].reuse('#{new_id}')
```
?

Reply:
I don't think you need to mess with AUTOCOMPLETERS.  I think just `$('##{first_id}').reuse('#{new_id}')`.  But I'd have to test it, too! :)